src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -1020,9 +1399,40 @@ static void verify(CSharpCompilation comp)
             }
         }
 
-        // PROTOTYPE: nint and nuint should be allowed.
         [Fact]
-        public void MemberName()
+        public void MemberName_01()
+        {
+            var source =
+@"namespace N
+{
+    class nint { }
+    class Program
+    {
+        internal static object nuint;
+        static void Main()
+        {
+            _ = new nint();
+            _ = new @nint();
```

Comment:
>new @nint(); [](start = 16, length = 12)

Unless we already have a test verifying this, consider testing that this and the previous line bind to the same type.

Reply:
And that type is the N.nint class 

---
In reply to: [392543787](https://github.com/dotnet/roslyn/pull/42360#discussion_r392543787) [](ancestors = 392543787)